### PR TITLE
 Fixes #116 advanced search not working properly

### LINKED ISSUE #116: Advanced Search not working properly.
Under advanced search, clicking on the check boxes does nothing. Automatically, it gets unchecked. Screenshots -

selection_014

Unchecked after few secs.
selection_015

UPDATE - This problem occurs under the following headings - Providers, Authors, Collections

### PR DESCRIPTION
**Pull request by nikhilrayaprolu**
image

fixed it where the selected one is checked.

### CODE REVIEW

File: src/app/advancedsearch/advancedsearch.component.ts [hunk 1]
```diff
@@ -12,9 +12,16 @@ import {Observable} from 'rxjs';
...
+  selectedelements:Array<any>=[]
+  changeurl(modifier,element) {
 
     this.querylook['query'] = this.querylook['query'] + '+' + decodeURIComponent(modifier);
+    this.selectedelements.push(element)
+    this.route.navigate(['/search'], {queryParams: this.querylook});
+  }
+  removeurl(modifier){
+    console.log('print')
+    this.querylook['query']=this.querylook['query'].replace('+' + decodeURIComponent(modifier),'')
```
Comment (harshit98) [inline src/app/advancedsearch/advancedsearch.component.ts]: @nikhilrayaprolu previously an issue no #168 was raised and here is the fix for that #178 . Please avoid `console.log`

Comment (nikhilrayaprolu) [inline src/app/advancedsearch/advancedsearch.component.ts]: done @harshit98 .

**Comment by mariobehling** [comment]
Please start to follow best practices now: http://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia/#commits

Add link to issue in PR description.

**Comment by mariobehling** [comment]
Please provide a link to a test page.

**Comment by harshit98** [comment]
@nikhilrayaprolu For generating a preview link, I have already added a deploy script in `package.json` I can see, this PR has branch `advancedsearchfix` please do the following -
- `git checkout advancedsearchfix`
- `ng build`
- `npm run deploy`

and then go to `https://yourusername.github.io/susper.com` done! :smile:

**Comment by mariobehling** [comment]
@harshit98 Can we do, what is possible in other projects, e.g. loklak? Please provide a preview link with every PR. You should do that anyways in order to test your results on the net.

**Comment by nikhilrayaprolu** [comment]
@mariobehling deployed the pull at http://susper-pr-225.herokuapp.com/

**Comment by harshit98** [comment]
@nikhilrayaprolu Please squash your commits.

**Comment by nikhilrayaprolu** [comment]
@harshit98 @daminisatya squashed the commits.

**Pull request merged**